### PR TITLE
Redesign the passive acquisition dialog

### PR DESCRIPTION
**Pull request by alesgenova**
Related to #1511 

### New Dialog
screenshot from 2018-05-24 13-38-31

### Custom Filename Format
screenshot from 2018-05-24 13-42-11

### Advanced Filename Format
screenshot from 2018-05-24 13-41-09

### Wrong Filename Format
screenshot from 2018-05-24 13-40-22

### CODE REVIEW

File: tomviz/acquisition/MatchInfo.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace tomviz {
+
+struct CapGroup {
+  CapGroup(QString name_, QString text) : name(name_), capturedText(text) {}
+  QString name;
+  QString capturedText;
+};
+
+struct MatchInfo {
+  bool matched;
```
Comment (cjh1) [inline tomviz/acquisition/MatchInfo.h]: Why do we need this? If groups contains items then we know the regex matched?

File: tomviz/acquisition/BasicFormatWidget.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  BasicFormatWidget(QWidget* parent = nullptr);
+  ~BasicFormatWidget() override;
+
+  QString getRegex() const;
+  QString getPythonRegex() const;
+  MatchInfo matchFileName(QString) const;
+  QJsonArray getRegexGroups() const;
+  QJsonObject getRegexSubsitutions() const;
+
+public slots:
```
Comment (cryos) [inline tomviz/acquisition/BasicFormatWidget.h]: Don't need empty public slots, just remove this and the empty lines

File: tomviz/acquisition/CustomFormatWidget.cxx [hunk 3]
```diff
@@ -0,0 +1,94 @@
+#include "CustomFormatWidget.h"
```
Comment (cryos) [inline tomviz/acquisition/CustomFormatWidget.cxx]: Missing copyright header

File: tomviz/acquisition/CustomFormatWidget.h [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+class CustomFormatWidget : public QWidget
+{
+  Q_OBJECT
+
+public:
+  CustomFormatWidget(QWidget* parent = nullptr);
+  ~CustomFormatWidget() override;
+  QStringList getFields() const;
+  void setFields(QString, QString, QString, QString, QString);
```
Comment (cryos) [inline tomviz/acquisition/CustomFormatWidget.h]: Why isn't this a QStringList?

File: tomviz/acquisition/CustomFormatWidget.h [hunk 5]
```diff
@@ -0,0 +1,45 @@
+#ifndef tomvizCustomFormatWidget_h
```
Comment (cryos) [inline tomviz/acquisition/CustomFormatWidget.h]: Missing copyright header

File: tomviz/acquisition/MatchInfo.h [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#ifndef tomvizMatchInfo_h
```
Comment (cryos) [inline tomviz/acquisition/MatchInfo.h]: Missing copyright header

File: tomviz/acquisition/MatchInfo.h [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+#define tomvizMatchInfo_h
+
+#include <QList>
+
+namespace tomviz {
+
+struct CapGroup
+{
+  CapGroup(QString name_, QString text) : name(name_), capturedText(text) {}
```
Comment (cryos) [inline tomviz/acquisition/MatchInfo.h]: Why not const ref for name/text?

File: tomviz/acquisition/PassiveAcquisitionWidget.h [hunk 8]
```diff
@@ -16,11 +16,15 @@
 #ifndef tomvizPassiveAcquisitionWidget_h
 #define tomvizPassiveAcquisitionWidget_h
 
+#include <QDialog>
+
+#include "MatchInfo.h"
+
 #include <QLabel>
+#include <QMap>
```
Comment (cryos) [inline tomviz/acquisition/PassiveAcquisitionWidget.h]: Not sure why this was added, don'

**Comment by cryos** [comment]
I would still like to see a sample file for the default file name format, instead it asks me to paste a sample. I tried my_p4.dm3 and it didn't do anything, I am not sure how it is supposed to run in the passive data acquisition dialog.

File: tomviz/acquisition/PassiveAcquisitionWidget.cxx [hunk 9]
```diff
@@ -444,38 +435,89 @@ void PassiveAcquisitionWidget::checkEnableWatchButton()
...
+  validateTestFileName();
 }
 
-void PassiveAcquisitionWidget::stopWatching()
+void PassiveAcquisitionWidget::onRegexChanged(QString regex)
 {
-  m_watchTimer->stop();
-  m_ui->stopWatchingButton->setEnabled(false);
-  m_ui->watchButton->setEnabled(true);
+  qDebug() << regex;
```
Comment (cryos) [inline tomviz/acquisition/PassiveAcquisitionWidget.cxx]: This looks like leftover debug

**Comment by cryos** [comment]
Looks better with the latest commit, but still surprising that when I open the dialog it is populated, the text entry is green, but when I make changes it turns an orange color and doesn't show me the result in the table below. Only custom seems to be evaluating the string as I change it.

**Comment by alesgenova** [comment]
The orange color indicates that the filename doesn't match the current regex, and as a consequence the table has no captured text to show. Each time the filename changes, the match is reevaluated and UI components should be updated as needed.
I think that the example you provided ("my_p4.dm3") doesn't match the default options because it expects something like "my_p4degree.dm3". Let me know if that's not the case, and I can dig deeper

**Comment by cryos** [comment]
I see now, I don't see a good reason to add the required suffix degree to dm3 and not tiff. Better that neither need the suffix degree, but that now makes more sense. I think our default should just be <prefix>[n|p]<angle><suffix>.*. We have a bit of space on this, and I think I would rather have a file name format, or an extension combo box. Switching to custom could just grey out the suffix.

We could then just offer a couple of format, and the extensions which are much easier to see when specified separately. We could also just hide the tif[f] by offering TIFF, DM3, etc as valid extensions we support.

**Pull request merged**